testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/providers/JsonBindingTest.java

```diff
@@ -0,0 +1,35 @@
+package org.jboss.resteasy.test.providers;
+
+import org.jboss.resteasy.plugins.providers.jsonb.JsonBindingProvider;
+import org.jboss.resteasy.plugins.server.servlet.ResteasyContextParameters;
+import javax.ws.rs.core.MediaType;
+import org.junit.Test;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Check when ResteasyConfiguration.class is null in JsonBindingProvider
+ * system property is checked and the property is set accordingly.
+ * There are integration tests that check the properly setting when
+ * a ResteasyConfiguration.class is provided.
+ */
+public class JsonBindingTest {
+    private MediaType mediaType = new MediaType("application", "json");
+
+    @Test
+    public void testUseJSONB() throws Exception
+    {
+        System.setProperty(ResteasyContextParameters.RESTEASY_PREFER_JACKSON_OVER_JSONB, "true");
```

Comment:
@rsearls this is fine, however you need to be sure whatever was set for this system property before the test is restored after it. IOW, the test should leave the environment in the same state it was before executing, otherwise all tests running after this would be affected.